tools/autograd/templates/Functions.cpp

```diff
@@ -1387,9 +1387,9 @@ Tensor as_strided_backward(Tensor grad, TensorGeometry input_geometry, IntList s
   //       more sensible to raise here.
 
   // Step (1): create underlying tensor as "storage"
-  auto shared_offset = std::min(input_geometry.storage_offset(), storage_offset);
+  auto shared_offset = std::min(input_geometry.storage_offset(), *storage_offset);
```

Comment:
I am not entirely sure here, is the `Tensor grad` storage_offset exactly the same as the tensor's storage offset for every case? so that the check/set defaults here should be just applied to the defaults of the gradient tensor?

Reply:
Or in general, Since it's in the autograd stage, we already have the storage_offset set in the forward pass and recorded those variables, do we really need to set the defaults here?

Reply:
This is indeed a problem.

I wrote this function so I hope I can clarify a bit. There used to be two `as_strided`, one with `storage_offset` and one without. The second just figures out what value `storage_offset` should be and calls the first. This backward is defined on the first one. So I would suggest, 
1. keep `storage_offset` as non optional here.
2. in `derivatives.yaml`, when writing last backward argument, use `storage_offset_.value_or(self.storage_offset())`.

Alternatively, just keep the old code structure.